src/inmanta/agent/agent.py

```diff
@@ -800,14 +810,17 @@ async def get_facts(self, resource: JsonType) -> Apireturn:
             return await executor.get_facts(resource_refs[0])
 
     async def setup_executor(
-        self, version: int, action: const.ResourceAction, resources: list[JsonType]
+        self, version: int, action: const.ResourceAction, resources: list[JsonType], resource_types: set[ResourceType]
     ) -> tuple[dict[ResourceVersionIdStr, const.ResourceState], list[ResourceDetails], Optional[executor.Executor]]:
-        # TODO refactor wiring of tuples around -> not sure how
-
         """
         Setup an executor for resource deployment
 
-        returns a tuple
+        :param version: model version being deployed
+        :param action: the resource action associated with this deploy
+        :param resources: list of resources in serialized form: a dict with information about
+            this resource and its desired state
+        :param resource_types: set of all resource types composing this version of the model
```

Comment:
```suggestion
        :param resource_types: set of all resource types composing this version of the model for this agent, even if they are not being deployed right now.
```
Tried to make this a bit more explicit, wdyt?